src/python/grpcio/grpc/_channel.py

```diff
@@ -652,16 +663,24 @@ def __init__(self, channel):
     self.managed_calls = None
 
 
-def _call_spin(state):
-  while True:
-    event = state.completion_queue.poll()
-    completed_call = event.tag(event)
-    if completed_call is not None:
-      with state.lock:
-        state.managed_calls.remove(completed_call)
-        if not state.managed_calls:
-          state.managed_calls = None
-          return
+def _run_call_spin_thread(state):
+  call_spin_running = True
+  def call_spin():
+    while call_spin_running:
+      event = state.completion_queue.poll()
+      completed_call = event.tag(event)
+      if completed_call is not None:
+        with state.lock:
+          state.managed_calls.remove(completed_call)
+          if not state.managed_calls:
+            state.managed_calls = None
+            return
+  def stop_call_spin(timeout):
+    call_spin_running = False
```

Comment:
I realized that there is no point in trying to join this thread.  ```completion_queue.poll()``` can be interrupted by a sigint, but those are only caught on the main thread.  There is currently no backdoor to abort a poll on a child thread.  I'll put in a fix for that.